### PR TITLE
Use official callbag types

### PR DESCRIPTION
**Pull request by jvanbruegge**
@staltz what do you think? No manually specifying any, and minimal changes to types in tests

### CODE REVIEW

File: src/pipe.ts [hunk 1]
```diff
@@ -1,95 +1,95 @@
-import { Operator, Source, Sink, Callbag } from './types';
+import { Operator, Source, Sink } from './types';
 
 // Returning a new callbag
 
 // prettier-ignore
-export function pipe<A, B>(source: Source<A> | Callbag<A>, o1: Operator<A, B>): Callbag<B>;
```
Comment (jvanbruegge) [inline src/pipe.ts:6]: Removing the Callbag here was also done in your PR, why? I added it so `subject` could be used in `pipe`.

Comment (staltz) [reply src/pipe.ts:6]: Oh, I didn't realize that `subject` could be in `pipe`, good point!

**Comment by staltz** [comment]
@jvanbruegge Could you merge my PR #6 and then add additional fixes in other commits instead of just redoing my work? I have another PR that builds on #6. It's weird to make a PR competing with mine instead of cooperating on that one.

**Comment by jvanbruegge** [comment]
That would not really change things, does it? The rebase would still be the same no matter if I undo all your changes in your PR or just make a new one without them.

I will change back the pipe types to work with subject later when I get back home

**Comment by staltz** [comment]
I mean, I wouldn't bother if you added commits to branch `callbag-types` instead of making a new branch. Notice also that there are updates such as 

```diff
-       of([1, 2, 3, 4, 5].map(of)),
+       of(...[1, 2, 3, 4, 5].map(x => of(x))),
```

Although your types make it nice that we don't have to update `(t: any, d: any) =>`, the benefit of my type updates is that they caught the `of()` case above as faulty.

**Pull request closed without merging**